### PR TITLE
Fix/display channel name taken error

### PR DESCRIPTION
**Pull request by siepra**

### CODE REVIEW

File: packages/frontend/src/renderer/components/widgets/channels/CreateChannel/CreateChannel.tsx [hunk 1]
```diff
@@ -65,7 +65,11 @@ const useStyles = makeStyles(theme => ({
 }))
 
 export const parseChannelName = (name = '') => {
-  return name.toLowerCase().replace(/ +/g, '-')
+  return name
+    .toLowerCase()
+    .trimStart()
```
Comment (EmiM) [inline packages/frontend/src/renderer/components/widgets/channels/CreateChannel/CreateChannel.tsx:70]: `trim` does the same as trimStart + trimEnd

File: packages/frontend/src/renderer/containers/widgets/channels/CreateChannel.tsx [hunk 2]
```diff
@@ -15,10 +25,16 @@ export const CreateChannel = () => {
   const community = useSelector(communities.selectors.currentCommunityId)
   const channels = useSelector(publicChannels.selectors.publicChannels)
 
+  const communityErrors = useSelector(errors.selectors.currentCommunityErrors)
+  const error = communityErrors?.find(error => error.type === SocketActionTypes.CREATED_CHANNEL)
```
Comment (EmiM) [inline packages/frontend/src/renderer/containers/widgets/channels/CreateChannel.tsx:29]: Can't we have a selector for this (like "get by type")? This way one doesn't have to filter manually by error.type every time

File: packages/frontend/src/renderer/containers/widgets/channels/CreateChannel.tsx [hunk 3]
```diff
@@ -31,6 +47,25 @@ export const CreateChannel = () => {
...
 
   const createChannel = (name: string) => {
+    // Clear errors
+    if (error) {
+      dispatch(
+        errors.actions.clearError(error)
+      )
+    }
+    // Validate channel name
+    if (channels.some(channel => channel.name === name)) {
```
Comment (EmiM) [inline packages/frontend/src/renderer/containers/widgets/channels/CreateChannel.tsx:57]: I'm thinking if maybe we should validate it on waggle side as we are doing it with checking duplication of username.

**Review by vinkabuki — APPROVED** [review]

File: packages/frontend/src/renderer/containers/widgets/channels/CreateChannel.tsx (hunk 3 above)
Comment (vinkabuki) [reply packages/frontend/src/renderer/containers/widgets/channels/CreateChannel.tsx:57]: What are arguments behind this? Checking username duplication is different - it's owner who checks for it, and he is the one who always knows about every taken username, and registration module is in waggle. This case is different and checking it in waggle doesn't make things safer or better imo

File: packages/frontend/src/renderer/components/widgets/channels/CreateChannel/CreateChannel.tsx (hunk 1 above)
Comment (siepra) [reply packages/frontend/src/renderer/components/widgets/channels/CreateChannel/CreateChannel.tsx:70]: I've been trying to use it and it turned out it trims also the spaces between the words

File: packages/frontend/src/renderer/containers/widgets/channels/CreateChannel.tsx (hunk 3 above)
Comment (siepra) [reply packages/frontend/src/renderer/containers/widgets/channels/CreateChannel.tsx:57]: So what I was thinking when doing it this way was that the main purpose of the task is to present proper validation error message (and improve UX at this stage). We've been through a lot of discussions about uniqueness of channel's names/addresses and I think we haven't come with any ideal solution for now. 
What I did is relatively simple and it provides good user experience as it displays understandable error message when trying to add a channel with the same name as one of those which are already in our store, or subscribes to the channel we haven't yet subscribed to

File: packages/frontend/src/renderer/components/widgets/channels/CreateChannel/CreateChannel.tsx (hunk 1 above)
Comment (EmiM) [reply packages/frontend/src/renderer/components/widgets/channels/CreateChannel/CreateChannel.tsx:70]: It doesn't. I use it in a different place and also check the docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim

**Pull request merged**